unit_test/test_spdm_callback/spdm_cert_verify_callback.c

```diff
@@ -0,0 +1,253 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2024 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#include "spdm_unit_test.h"
+#include "library/spdm_common_lib.h"
+#include "spdm_crypt_ext_lib/spdm_crypt_ext_lib.h"
+
+/**
+ * tcg-dice-TcbInfo OID: 2.23.133.5.4.1
+ * https://trustedcomputinggroup.org/wp-content/uploads/DICE-Attestation-Architecture-Version-1.1-Revision-18_pub.pdf
+ **/
+uint8_t m_libspdm_tcg_dice_tcbinfo_oid[] = {0x67, 0x81, 0x05, 0x05, 0x04, 0x01};
+
+bool m_libspdm_must_have_dice_tcb_info = false;
+
+/*reference DiceTcbinfo*/
+
+/*vendor: INTC*/
+uint8_t m_libspdm_dice_tcbinfo_vendor[] = {0x49, 0x4E, 0x54, 0x43};
+/*model: S3M GNR*/
+uint8_t m_libspdm_dice_tcbinfo_model[] = {0x53, 0x33, 0x4D, 0x20, 0x47, 0x4E, 0x52};
+/*version: 000200000000008B*/
+uint8_t m_libspdm_dice_tcbinfo_version[] = {0x30, 0x30, 0x30, 0x32, 0x30, 0x30, 0x30, 0x30, 0x30,
+                                            0x30, 0x30, 0x30, 0x30, 0x30, 0x38, 0x42};
+/*svn*/
+uint8_t m_libspdm_dice_tcbinfo_svn[] = {0x01};
+/*layer*/
+uint8_t m_libspdm_dice_tcbinfo_layer[] = {0x01};
+/*fwids*/
+uint8_t m_libspdm_dice_tcbinfo_fwids[] = {0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04,
+                                          0x02, 0x02, 0x04, 0x30, 0x6B, 0x44, 0x7B, 0x5E, 0x99,
+                                          0x21, 0x0A, 0x58, 0x8A, 0x7B, 0x31, 0x7D, 0xBA, 0x2D,
+                                          0x4A, 0x7F, 0x75, 0xE6, 0x97, 0xF2, 0x07, 0xE0, 0xC2,
+                                          0x99, 0x78, 0xF3, 0xF6, 0x2B, 0x53, 0xF5, 0xBE, 0xEB,
+                                          0x73, 0xF0, 0x37, 0xB8, 0x79, 0xC1, 0xFF, 0x76, 0x2A,
+                                          0x3A, 0x39, 0xCA, 0xE2, 0x8C, 0xF0, 0x56};
+/*type*/
+uint8_t m_libspdm_dice_tcbinfo_type[] = {0x46, 0x69, 0x72, 0x6D, 0x77, 0x61, 0x72, 0x65, 0x20,
+                                         0x44, 0x69, 0x67, 0x65, 0x73, 0x74};
+
+/*verify cert DiceTcbInfo extension*/
+bool libspdm_verify_cert_dicetcbinfo(const void *cert, size_t cert_size) {
+    bool result;
+    uint8_t spdm_dice_tcb_info[256];
+    size_t spdm_dice_tcb_info_size;
+    uint8_t *ptr;
+    int32_t length;
+    size_t obj_len;
+    uint8_t *end;
+
```

Comment:
General: we can't assume that the adopter code will provide reference values for all TCBinfo components hardcoded above (m_libspdm_dice_tcbinfo_xyz).  If a reference is not provided, this function should skip mem_equal() for that component.

Reply:
Yes.  The code assume only follow reference will be provided. And the other components will be skip even if existing in cert.

```
/*vendor: INTC*/
/*model: S3M GNR*/
/*version: 000200000000008B*/
/*svn*/
/*layer*/
/*fwids*/
/*type*/
```

Reply:
I was saying that the code should not require hardcoded references of these components (vendor, model, version, etc.).  For example, the verifier can choose to hardcode reference for only "version" because the verifier only cares about version.

Reply:
Please note, this test is only reference implementation, to demonstrate how to do the alias cert verification.

The integrator should decide what to do - input as parameter, or hardcode somewhere.